dispatch/static/manager/src/js/components/fields/WidgetField.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react'
+import R from 'ramda'
+import { connect } from 'react-redux'
+
+import Panel from '../Panel'
+import WidgetSelectInput from '../inputs/WidgetSelectInput'
+import WidgetFieldWrapper from '../ZoneEditor/WidgetField'
+
+class WidgetFieldComponent extends React.Component {
+  handleWidgetChange(widgetId) {
+    const data = R.merge(this.props.data || {}, {
+      id: widgetId,
+      data: this.props.data || {}
+    })
+
+    this.props.onChange(data)
```

Comment:
Could this method be re-written like this?

```js
this.props.onChange({
  id: widgetId,
  data: {}
})
```

In my opinion, changing the widget should clear the field data